dbadvisor-core/src/main/java/com/github/embeditcz/dbadvisor/core/internal/DbAdvisorReportGenerator.java

```diff
@@ -23,26 +23,64 @@
     private final IssueObjectMapper issueObjectMapper;
 
     public void generate() {
-        generateJsonReport();
+        List<Issue> issues = issueRepository.getIssues();
+        generateJsonReport(issues);
+        generateHtmlReport(issues);
     }
 
-    private void generateJsonReport() {
+    private void generateJsonReport(List<Issue> issues) {
         try {
-            File reportFile = resolveJsonReportFile();
-            List<Issue> issues = issueRepository.getIssues();
+            File reportFile = resolveReportFile(".json");
             issueObjectMapper.getObjectMapper().writeValue(reportFile, issues);
-            log.log(INFO, "The issues report is generated in " + reportFile);
+            log.log(INFO, "The issues json report is generated in " + reportFile);
         } catch (Exception e) {
             log.log(SEVERE, "Unable to write json report.", e);
         }
     }
 
-    private File resolveJsonReportFile() {
+    private void generateHtmlReport(List<Issue> issues) {
+        File reportFile = resolveReportFile(".html");
+        try (BufferedReader templateReader = createHtmlReportTemplateReader()) {
+            try (PrintWriter reportWriter = createHtmlReportWriter(reportFile)) {
+                templateReader.lines().forEach(line -> processTemplateLine(line, reportWriter, issues));
+                reportWriter.flush();
+                log.log(INFO, "The issues html report is generated in " + reportFile);
+            }
+        } catch (Exception e) {
+            log.log(SEVERE, "Unable to write html report.", e);
+        }
+    }
+
+    private PrintWriter createHtmlReportWriter(File reportFile) throws IOException {
+        return new PrintWriter(new BufferedWriter(new FileWriter(reportFile)));
+    }
+
+    private BufferedReader createHtmlReportTemplateReader() {
+        InputStream templateInputStream = getClass().getClassLoader().getResourceAsStream("report-template.html");
+        return new BufferedReader(new InputStreamReader(templateInputStream));
+    }
+
+    private void processTemplateLine(String line, PrintWriter reportWriter, List<Issue> issues) {
+        if (line.equals("    let data = [];")) {
```

Comment:
Isn't it better to trim the line and then compare also to trimmed constant? Just to prevent code reformatting.

Reply:
Just an idea: we can put to file something like:
let data = []; //injected data
And simply replace "[]; //injected data" with json.
WDYT?

Reply:
as discussed offline, added trimming